"error calling Constructor for CustomType" exception when CustomPropertyComparator is registered for value's parent class

`error calling Constructor for org.javers.core.metamodel.type.CustomType` exception is thrown on `javers.compare(..)` when there is a CustomPropertyComparator registered for a parent class of some value that exist in objects that are being compared.

Here is a test case that reproduces the issue: https://github.com/kkonstantin42/javers/blob/custom-property-comparator-with-inheritance/javers-core/src/test/groovy/org/javers/core/cases/CustomPropertyComparatorWithInheritanceCase.groovy

This behaviour has appeared after CustomType constructor was changed to accept CustomPropertyComparator as a second parameter in this commit: https://github.com/javers/javers/commit/ec6113bb160d975b1733cd6089d5c3b5f31b7d7c#diff-461b37401e91dc6db100811e06995807
Possibly there should be a special case in TypeFactory.spawnFromPrototype method that would handle creation of CustomType objects. Can submit a PR.

Here is a stacktrace:
```
java.lang.RuntimeException: error calling Constructor for org.javers.core.metamodel.type.CustomType

	at org.javers.core.metamodel.type.JaversType.spawn(JaversType.java:50)
	at org.javers.core.metamodel.type.TypeFactory.spawnFromPrototype(TypeFactory.java:121)
	at org.javers.core.metamodel.type.TypeFactory.infer(TypeFactory.java:74)
	at org.javers.core.metamodel.type.TypeMapperState.infer(TypeMapperState.java:157)
	at org.javers.core.metamodel.type.TypeMapperState.lambda$getJaversType$0(TypeMapperState.java:89)
	at org.javers.core.metamodel.type.TypeMapperState.computeIfAbsent(TypeMapperState.java:113)
	at org.javers.core.metamodel.type.TypeMapperState.getJaversType(TypeMapperState.java:89)
	at org.javers.core.metamodel.type.TypeMapper.getJaversType(TypeMapper.java:129)
	at org.javers.core.metamodel.type.ManagedClassFactory.lambda$null$2(ManagedClassFactory.java:54)
	at org.javers.core.metamodel.type.JaversProperty.getType(JaversProperty.java:23)
	at org.javers.core.graph.ObjectGraphBuilder.lambda$getSingleReferencesWithManagedTypes$0(ObjectGraphBuilder.java:105)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:174)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at org.javers.common.collections.Lists.positiveFilter(Lists.java:71)
	at org.javers.core.metamodel.type.ManagedClass.getManagedProperties(ManagedClass.java:72)
	at org.javers.core.metamodel.type.ManagedType.getProperties(ManagedType.java:55)
	at org.javers.core.graph.ObjectGraphBuilder.getSingleReferencesWithManagedTypes(ObjectGraphBuilder.java:105)
	at org.javers.core.graph.ObjectGraphBuilder.buildSingleEdges(ObjectGraphBuilder.java:75)
	at org.javers.core.graph.ObjectGraphBuilder.buildEdges(ObjectGraphBuilder.java:70)
	at org.javers.core.graph.ObjectGraphBuilder.buildGraphFromCdo(ObjectGraphBuilder.java:58)
	at org.javers.core.graph.ObjectGraphBuilder.buildGraph(ObjectGraphBuilder.java:47)
	at org.javers.core.graph.LiveGraphFactory.createLiveGraph(LiveGraphFactory.java:39)
	at org.javers.core.diff.DiffFactory.buildGraph(DiffFactory.java:99)
	at org.javers.core.diff.DiffFactory.compare(DiffFactory.java:54)
	at org.javers.core.JaversCore.compare(JaversCore.java:173)
	at org.javers.core.cases.CustomPropertyComparatorWithInheritanceCase.should use CustomPropertyComparator for all subclasses(CustomPropertyComparatorWithInheritanceCase.groovy:65)
Caused by: java.lang.NoSuchMethodException: org.javers.core.metamodel.type.CustomType.<init>(java.lang.reflect.Type)
	at java.lang.Class.getConstructor0(Class.java:3082)
	at java.lang.Class.getConstructor(Class.java:1825)
	at org.javers.core.metamodel.type.JaversType.spawn(JaversType.java:47)
	... 30 more
```

--- a/javers-core/src/main/java/org/javers/core/metamodel/type/CustomType.java
+++ b/javers-core/src/main/java/org/javers/core/metamodel/type/CustomType.java
@@ -36,4 +36,8 @@ public class CustomType<T> extends ClassType {
     public boolean equals(Object left, Object right) {
         return comparator.equals((T)left, (T)right);
     }
+
+    public CustomPropertyComparator<T, ?> getComparator() {
+        return comparator;
+    }
 }

--- a/javers-core/src/main/java/org/javers/core/metamodel/type/TypeFactory.java
+++ b/javers-core/src/main/java/org/javers/core/metamodel/type/TypeFactory.java
@@ -114,6 +114,9 @@ class TypeFactory {
             ManagedClass managedClass = managedClassFactory.createFromPrototype(javaRichType.javaClass, javaRichType.getScan(),
                     managedPrototype.getManagedClass().getManagedPropertiesFilter());
             return managedPrototype.spawn(managedClass, javaRichType.getScan().typeName());
+        } else if (prototype instanceof CustomType) {
+            CustomType customTypePrototype = (CustomType) prototype;
+            return new CustomType(customTypePrototype.getBaseJavaType(), customTypePrototype.getComparator());
         } else {
             return prototype.spawn(javaRichType.javaType); //delegate to simple constructor
         }

--- a/javers-core/src/test/groovy/org/javers/core/cases/CustomPropertyComparatorWithInheritanceCase.groovy
+++ b/javers-core/src/test/groovy/org/javers/core/cases/CustomPropertyComparatorWithInheritanceCase.groovy
@@ -1,0 +1,71 @@
+package org.javers.core.cases
+
+
+import org.javers.core.JaversBuilder
+import org.javers.core.diff.changetype.ValueChange
+import org.javers.core.diff.custom.CustomPropertyComparator
+import org.javers.core.metamodel.object.GlobalId
+import org.javers.core.metamodel.property.Property
+import spock.lang.Specification
+
+/**
+ * @author konstantinkastanov* created on 2019-02-19
+ */
+class CustomPropertyComparatorWithInheritanceCase extends Specification {
+
+	class MightyWizard {
+		private IntValue level
+	}
+
+	class SimpleValue<T> {
+		private T value
+
+		SimpleValue(T value = null) {
+			this.value = value
+		}
+
+		T getValue() {
+			return value
+		}
+
+		void setValue(T value) {
+			this.value = value
+		}
+	}
+
+	class IntValue extends SimpleValue<Integer> {
+		IntValue(Integer value = null) {
+			super(value)
+		}
+	}
+
+	class SimpleValueComparator implements CustomPropertyComparator<SimpleValue, ValueChange> {
+		Optional<ValueChange> compare(SimpleValue left, SimpleValue right, GlobalId affectedId, Property property) {
+			if (left.getValue().equals(right.getValue()))
+				return Optional.empty()
+			return Optional.of(new ValueChange(affectedId, property.name, left.getValue(), right.getValue()));
+		}
+
+		@Override
+		boolean equals(SimpleValue left, SimpleValue right) {
+			return left.getValue().equals(right.getValue())
+		}
+	}
+
+	def "should use CustomPropertyComparator for all subclasses"(){
+		given:
+		def javers = JaversBuilder.javers()
+				.registerCustomComparator(new SimpleValueComparator(), SimpleValue).build()
+
+		when:
+		MightyWizard w1 = new MightyWizard();
+		w1.level = new IntValue(1)
+		MightyWizard w2 = new MightyWizard();
+		w2.level = new IntValue(2)
+		def diff = javers.compare(w1, w2)
+		then:
+		diff.changes.size() == 1
+	}
+
+
+}

